backend/src/StamAcasa.Common/Queue/NotificationQueueRabbit.cs

```diff
@@ -0,0 +1,68 @@
+﻿using System;
+using System.Threading.Tasks;
+using RabbitMQ.Client;
+using StamAcasa.Common.Notifications;
+using StamAcasa.Common.Services;
+
+namespace StamAcasa.Common.Queue
+{
+    public class NotificationQueueRabbit : IQueue
+    {
+        private string HostName { get; }
+
+        private RabbitConfig Config { get; }
+
+        private IConnection Connection { get; }
+
+        public NotificationQueueRabbit(string hostName, RabbitConfig config)
```

Comment:
hostName de ce nu poate fi in RabbitConfig?

Reply:
If we place hostName inside the Config object, then we need to place the connection inside as well, because hostName and connection represent the same concept, a Connection (hostname is used to create a connection).

But IMO, connection doesn't fit well inside the Config object, hence I opted for this separation.

Reply:
Inteleg ce zici, dar hostname este o cale catre instanta de rabbitmq. Connection ramane unde este. Nu vad legatura dintre IConnection si un string care reprezinta unde se afla instanta de rabbit mq.

Reply:
Raspund si eu in romana ... credeam ca trebuie sa lasam comments in engleza.

Din sugestiile tale initiale, am inteles ca trebuie sa avem doua moduri de constructie pentru NotificationQueueRabbit:
1. cu un obiect Config
2. cu o conexiune existenta, plus alti parametri, probabil tot un obiect Config

Daca mergem pe modelul in care hostName este in Config, cand vom folosi o conexiune existenta (sugestia 2) si vom construi un NotificationQueueRabbit, va trebui sa-i trimitem un Config care are hostName null, deci care este incomplet/invalid.

Daca nu te-am convins, sugerez sa mai vina un reviewer si sa procedam cum decide el. Oricum eu nu am inteles cum vrei sa procedez mai departe. Cum ar arata constructorul/constructorii lui NotificationQueueRabbit in varianta ta?

Reply:
Salut am logat un issue nou pt interfata cu RabbitMQ: https://github.com/code4romania/stam-acasa/issues/105
As vrea ca interfata construita in acel issue sa fie utilizata de toti clientii cozii, indiferent ca sunt producers sau consumers - in felul asta avem o abordare unified si fara unexpected bugs.